### PR TITLE
fix(mechanics): Surveillance personality no longer overrides staying

### PR DESCRIPTION
**Pull request by warp-core**
**Bugfix:**

Thank you to all the people who have reported the issue with the Winds of Light not actually staying in Peragenor, both on Steam and Discord.

## Fix Details
It turns out that AI::DoSurveillance() was setting jump and land commands on ships without checking if they were `staying` and so should not receive such commands. I do not know how long this has been the case, but it does happen in 0.9.14.
The issue was only discovered relatively recently where the Winds of Light, the Starling you are required to scan in the missions: `Remnant: Keystone Research 5` (and 6) which has both `staying` and `surveillance` personalities.
The behaviour of it not staying is also, however, rather intermittent, but the frequency with which this behaviour is exhibited can be drastically increased by removing all StellarObjects from Peragenor (optionally including the wormhole). After doing this, it will quickly leave the system (if the wormhole is present, it will usually use this route, but it does sometimes jump to Stercutus via hyperdrive) after you enter. In my observations, it tends to then stay in the system it is in until you enter, at which point it will rapidly leave. I do not understand this behaviour, since it seems independent of the StellarObjects in the system, unlike the behaviour in the initial system. Regardless, a ship with `staying` should not leave its starting system, so subsequent behaviour is (hopefully) irrelevant.

## Testing Done
Take off with the appropriate mission active, go to Peragenor, see the ship leave before this PR, and not leave with this PR.

## Save File
This save file can be used to verify the bugfix. The bug will occur when using {{insert commit hash / version}}, and will not occur when using this branch's build.
[ThisTookTooLong ToFigureOut~3019-07-05.txt](https://github.com/endless-sky/endless-sky/files/9915108/ThisTookTooLong.ToFigureOut.3019-07-05.txt)
(Note: I have already applied the removal of all StellarObjects but the wormhole in Peragenor in the changes section of this save file.)

### CODE REVIEW

**Review by RisingLeaf — APPROVED** [review]
no real structure changes, so my approve is not really important.
The rest looks also fine though

**Comment by Hurleveur** [comment]
I feel like they should still drop the drone, and just not leave afterwards

**Comment by Zitchas** [comment]
I agree that dropping drones is kind of an important plot aspect for `surveillance`.

**Review by Hurleveur — APPROVED** [review]

File: source/AI.cpp [hunk 1]
```diff
@@ -2381,6 +2381,7 @@ void AI::DoSwarming(Ship &ship, Command &command, shared_ptr<Ship> &target)
 
 void AI::DoSurveillance(Ship &ship, Command &command, shared_ptr<Ship> &target) const
 {
+	bool isStaying = ship.GetPersonality().IsStaying();
```
Comment (quyykk) [inline source/AI.cpp:2384]:
```suggestion
	const bool isStaying = ship.GetPersonality().IsStaying();
```

**Comment by quyykk** [comment]
If there is nothing to scan anymore, staying surveillance ships now just stop and don't do anything which I guess looks a bit out of place? But that seems to have been MZ's intent.

**Review by quyykk — APPROVED** [review]

**Pull request merged**